From f9df65ce335f4c09d92ad1189fb13d16f818062f Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Fri, 26 May 2023 09:40:25 +0200
Subject: [PATCH 376/841] feat(stm32mp1-fdts): add sysram node for ssp

NOT_UPSTREAMABLE

Add the sysram node to define the supply used.
Mandatory to properly mask the supply during reset.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git b1e0636525760afc9b869adef45383a78ea84573 feat(stm32mp1-fdts): add sysram node for ssp]
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Change-Id: I615cc88361fa8108cdd532cc8a0cafbdd03754b0
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/308669
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 fdts/stm32mp13-ssp-bl2.dtsi | 6 ++++++
 fdts/stm32mp15-ssp-bl2.dtsi | 6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/fdts/stm32mp13-ssp-bl2.dtsi b/fdts/stm32mp13-ssp-bl2.dtsi
index 0307b9bea..bb86c4972 100644
--- a/fdts/stm32mp13-ssp-bl2.dtsi
+++ b/fdts/stm32mp13-ssp-bl2.dtsi
@@ -8,6 +8,12 @@
 		/delete-property/ mmc0;
 	};
 
+	sysram: sram@2ffe0000 {
+		compatible = "mmio-sram";
+		reg = <0x2ffe0000 0x20000>;
+		vddcore-supply = <&vddcore>;
+	};
+
 	soc {
 		efuse@5c005000 {
 			cfg0_otp: cfg0_otp@0 {
diff --git a/fdts/stm32mp15-ssp-bl2.dtsi b/fdts/stm32mp15-ssp-bl2.dtsi
index 8698a9be7..e2d5625e2 100644
--- a/fdts/stm32mp15-ssp-bl2.dtsi
+++ b/fdts/stm32mp15-ssp-bl2.dtsi
@@ -19,6 +19,12 @@
 	/delete-node/ cpu0_opp_table;
 	/delete-node/ psci;
 
+	sysram: sram@2ffc0000 {
+		compatible = "mmio-sram";
+		reg = <0x2ffc0000 0x40000>;
+		vddcore-supply = <&vddcore>;
+	};
+
 	soc {
 		efuse@5c005000 {
 			cfg2_otp: cfg2_otp@8 {
-- 
2.47.2

